ci/lint_platformio/check_root_platformio_lockdown.py

```
@@ -0,0 +1,264 @@
+    target_line_no: int, all_lines: Iterable[str]
+) -> tuple[bool, bool]:
+    """Look at the full new file for a `; justification:` and `; added-in:`
+    comment within 5 lines before or after `target_line_no` (1-indexed).
+
+    Returns (has_justification, has_added_in).
+    """
+    lines = list(all_lines)
+    lo = max(0, target_line_no - 1 - 5)
+    hi = min(len(lines), target_line_no - 1 + 6)
+    window = lines[lo:hi]
+    has_just = any(_JUSTIFICATION_RE.search(l) for l in window)
```

Ruff (0.15.17)

[error] Ambiguous variable name: `l`

(E741)